addons/account/models/account_bank_statement.py

```diff
@@ -828,6 +834,17 @@ def _compute_is_reconciled(self):
                 # The journal entry seems reconciled.
                 st_line.is_reconciled = True
 
+            # Compute residual amount
+            st_line_currency = suspense_lines.currency_id or suspense_lines.company_currency_id
+            balance_field, residual_field = ('amount_currency', 'amount_residual_currency') if suspense_lines.currency_id else ('balance', 'amount_residual')
+
+            if self.to_check:
+                st_line.amount_residual = -sum(liquidity_lines.mapped(balance_field))
```

Comment:
`st_line.amount_residual = st_line.amount_currency if st_line.foreign_currency_id else st_line.amount`
because in case of `to_check`, don't trust the journal items!

Reply:
EDIT: inverse the amount